### PR TITLE
feat: Implement site banner - EXO-65534 - Meeds-io/MIPs#72

### PR DESCRIPTION
**Pull request by GouadriaHanen**

### CODE REVIEW

File: component/api/src/main/java/org/exoplatform/portal/mop/service/LayoutService.java [hunk 1]
```diff
@@ -330,4 +332,16 @@ default void saveImportStatus(Status status) { // NOSONAR
   default List<PortalConfig> getSites(SiteFilter siteFilter) { // NOSONAR
     throw new UnsupportedOperationException();
   }
+
+  /**
+   * Retrieve site banner as stream
+   *
+   * @param  siteType {@link SiteType}
+   * @return                         found {@link InputStream}   *
```
Comment (azayati) [inline component/api/src/main/java/org/exoplatform/portal/mop/service/LayoutService.java]:
```suggestion
   * @return                         found {@link InputStream}
```
format that

File: component/portal/src/main/java/org/exoplatform/portal/mop/storage/SiteStorageImpl.java [hunk 2]
```diff
@@ -260,6 +280,17 @@ public List<PortalData> getSites(SiteFilter siteFilter) {
     return siteEntities.stream().map(this::buildPortalData).toList();
   }
 
+  @Override
+  public InputStream getSiteBannerAsStream(long fileId) {
```
Comment (azayati) [inline component/portal/src/main/java/org/exoplatform/portal/mop/storage/SiteStorageImpl.java]: this should be in layout service layer

File: component/portal/src/main/java/org/exoplatform/portal/mop/storage/SiteStorageImpl.java [hunk 3]
```diff
@@ -110,6 +122,10 @@ public void create(PortalConfig config) {
   public void create(PortalData config) {
     SiteEntity entity = new SiteEntity();
     buildSiteEntity(entity, config);
+    if (StringUtils.isNotBlank(config.getBannerUploadId())) {
```
Comment (azayati) [inline component/portal/src/main/java/org/exoplatform/portal/mop/storage/SiteStorageImpl.java:130]: this should be in layout service layer

File: component/api/src/main/java/org/exoplatform/portal/mop/service/LayoutService.java [hunk 4]
```diff
@@ -330,4 +332,16 @@ default void saveImportStatus(Status status) { // NOSONAR
...
+   * @throws ObjectNotFoundException When program not found or file attachment
+   *
+   */
+  default InputStream getSiteBannerStream(SiteKey siteKey) throws ObjectNotFoundException { // NOSONAR
```
Comment (azayati) [inline component/api/src/main/java/org/exoplatform/portal/mop/service/LayoutService.java]:
```suggestion
  default InputStream getSiteBanner(String siteName) throws ObjectNotFoundException { // NOSONAR
```

**Review by azayati — CHANGES_REQUESTED** [review]

File: component/api/src/main/java/org/exoplatform/portal/config/model/PortalConfig.java [hunk 5]
```diff
@@ -324,6 +330,22 @@ public void setDisplayOrder(int displayOrder) {
...
+    }
+
+    public long getBannerFileId() {
+        return bannerFileId;
+    }
+
+    public void setBannerFileId(long bannerFileId) {
+        this.bannerFileId = bannerFileId;
+    }
+
```
Comment (azayati) [inline component/api/src/main/java/org/exoplatform/portal/config/model/PortalConfig.java]: format code

File: component/api/src/main/java/org/exoplatform/portal/mop/service/LayoutService.java (hunk 1 above)
Comment (azayati) [reply component/api/src/main/java/org/exoplatform/portal/mop/service/LayoutService.java]: consider that review

File: component/api/src/main/java/org/exoplatform/portal/mop/storage/SiteStorage.java [hunk 6]
```diff
@@ -15,6 +15,7 @@
  */
 package org.exoplatform.portal.mop.storage;
 
+import java.io.InputStream;
```
Comment (azayati) [inline component/api/src/main/java/org/exoplatform/portal/mop/storage/SiteStorage.java]: restore that file if not needed

File: component/portal/src/main/java/org/exoplatform/portal/mop/storage/SiteStorageImpl.java [hunk 7]
```diff
@@ -23,19 +23,20 @@
...
 import java.io.IOException;
 import java.io.InputStream;
 import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (azayati) [inline component/portal/src/main/java/org/exoplatform/portal/mop/storage/SiteStorageImpl.java]: add explicit imports

File: component/portal/src/main/java/org/exoplatform/portal/mop/storage/SiteStorageImpl.java [hunk 8]
```diff
@@ -350,4 +374,34 @@ private void savePermissions(Long id, PortalData config) {
...
+      throw new IllegalArgumentException("uploadId is mandatory");
+    }
+    if (oldFileId != null && oldFileId != 0) {
+      fileService.deleteFile(oldFileId);
+    }
+    UploadResource uploadResource = uploadService.getUploadResource(uploadId);
+    if (uploadResource == null) {
+      throw new IllegalStateException("Can't find uploaded resource with id : " + uploadId);
+    }
+    try { // NOSONAR
```
Comment (azayati) [inline component/portal/src/main/java/org/exoplatform/portal/mop/storage/SiteStorageImpl.java]: why no sonar ?

File: component/api/src/main/java/org/exoplatform/portal/mop/service/LayoutService.java [hunk 9]
```diff
@@ -330,4 +333,28 @@ default void saveImportStatus(Status status) { // NOSONAR
...
+
+  /**
+   * Retrieve site banner as stream
+   *
+   * @param  siteName name of the portal site
+   * @return                         found {@link InputStream}   *
+   * @throws ObjectNotFoundException When program not found or file attachment
+   *
+   */
+  default InputStream getSiteBannerStream(String siteName) throws ObjectNotFoundException, IOException { // NOSONAR
```
Comment (azayati) [inline component/api/src/main/java/org/exoplatform/portal/mop/service/LayoutService.java]: why no sonar ?

File: component/api/src/main/java/org/exoplatform/portal/mop/service/LayoutService.java [hunk 10]
```diff
@@ -330,4 +333,28 @@ default void saveImportStatus(Status status) { // NOSONAR
...
+  }
+
+  /**
+   * remove site banner as stream
+   *
+   * @param  siteName name of the portal site
+   * @throws ObjectNotFoundException When program not found or file attachment
+   *
+   */
+  default void removeSiteBanner(String siteName) throws ObjectNotFoundException { // NOSONAR
```
Comment (azayati) [inline component/api/src/main/java/org/exoplatform/portal/mop/service/LayoutService.java]: why no sonar ?

File: web/portal/src/main/webapp/WEB-INF/conf/jdbc/configuration.xml [hunk 11]
```diff
@@ -220,4 +220,19 @@
       <type>org.exoplatform.portal.mop.storage.listener.NodeDescriptionListener</type>
     </component-plugin>    
   </external-component-plugins>
+  <external-component-plugins>
+    <target-component>org.exoplatform.commons.file.services.NameSpaceService</target-component>
```
Comment (azayati) [inline web/portal/src/main/webapp/WEB-INF/conf/jdbc/configuration.xml:224]: where it is used ?

**Review by azayati — CHANGES_REQUESTED** [review]

File: web/portal/src/main/webapp/WEB-INF/conf/jdbc/configuration.xml (hunk 11 above)
Comment (GouadriaHanen) [reply web/portal/src/main/webapp/WEB-INF/conf/jdbc/configuration.xml:224]: here for the save of the file :
https://github.com/Meeds-io/gatein-portal/pull/661/files#diff-03d0a49739ad35956fd2c4da054ba21c373e8abccee15bffc510d008b4a31254R398

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**